kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -183,6 +188,19 @@ void shouldPassThroughRecordUnchanged(KafkaCluster cluster, Topic topic) throws
         }
     }
 
+    @Test
+    void shouldFailFastWhenConnectWithSSLToPlainListener(KafkaCluster cluster) {
+        assertThatThrownBy(() -> {
+            try (var tester = kroxyliciousTester(proxy(cluster))) {
+                String bootstrap = tester.getBootstrapAddress();
+                String[] split = bootstrap.split(":");
+                try (SSLSocket socket = (SSLSocket) SSLContext.getDefault().getSocketFactory().createSocket(split[0], Integer.parseInt(split[1]))) {
```

Comment:
Is it worth setting SO_TIMEOUT to a suitable large value?  The motivation would be to avoid a production code defect leading to a difficult to diagnose test suite hang.